### ISSUE TITLE
A field over SimpleMenu (issue)

### ISSUE THREAD

**Issue by apoterenko** [material-components/material-components-web#1978]
## Bugs

There is an issue when using Field and SimpleMenu. The source of a problem is css rule:

```
.mdc-text-field {
    will-change: opacity, transform, color;  // A workaround is: will-change: unset;
```

### What MDC-Web Version are you using?

0.28.0

### What browser(s) is this bug affecting?

Chrome at least

### What are the steps to reproduce the bug?

![dom](https://user-images.githubusercontent.com/12325691/35048994-c0f4042a-fbaf-11e7-88de-aafd16ab8ceb.png)

### What is the expected behavior?

![without_bug](https://user-images.githubusercontent.com/12325691/35048803-2764e39c-fbaf-11e7-9520-7f68566d48cc.png)

### What is the actual behavior?

![bug](https://user-images.githubusercontent.com/12325691/35048732-f5fa616a-fbae-11e7-8cbf-a11caf9a29fd.png)

**Comment by acdvorak** [comment]
Thanks for filing this issue!

Could you possibly provide a min-viable [CodePen](https://codepen.io/traviskaufman/pen/pNQmRp) that exposes the problem?

Are you able to repro it in other browsers (Safari, Firefox, Edge)?

**Comment by apoterenko** [comment]
![mdc-1978](https://user-images.githubusercontent.com/12325691/35458007-d7274e80-02eb-11e8-8dbf-3d64b9066680.png)

Live version here
https://apoterenko.github.io/mdc-issue-1978

material-components-web 0.28.0

**Comment by kfranqueiro** [comment]
FWIW I attempted to make a [reduced test case](https://codepen.io/kfranqueiro/pen/JpgdYy) out of curiosity and it doesn't seem to reproduce this issue, which leaves 2 possibilities:

* Something changed since 0.28.0 that fixed it (possible, since we've done work on menu since then)
* Something more complex about this example is responsible for the problem

(Well, or the 3rd possibility is I'm missing something obvious.)

**Comment by acdvorak** [comment]
@kfranqueiro The reason [your reduced test case](https://codepen.io/kfranqueiro/pen/JpgdYy) didn't repro the issue is because you happened to pick a component (`mdc-button`) that does _not_ create its own stacking context, as opposed to `mdc-select` which _does_ :grinning:

Simply adding `will-change: opacity` to the `mdc-menu-anchor` element in your CodePen is enough to trigger the issue: https://codepen.io/acdvorak/pen/BYXZNj

See the ["stacking context" article on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) for more info.

The problem is essentially that you have two `mdc-select` elements on the page that A) are siblings in the DOM, B) each create their own stacking context, and C) have the same `z-index`. When two elements have the same stacking value, whichever comes later in the source order wins.

I'm not sure if we can fix this in a generic way in MDC... _Maybe_ we could hoist the menu element up to the `<body>` and give it a high `z-index`. However, that would create additional problems: We'd have to pick the "right" `z-index`, and re-position the menu on `scroll`, `resize`, and `orientationchange` events, which is expensive.

In short, I don't see any good solution that can support every use case :cry:

**Comment by apoterenko** [comment]
Guys, thank you for your time.
I have a workaround and you can close one. 
Anyway - thanks for this great product!
